thamos/lib.py

```diff
@@ -417,6 +374,56 @@ def image_analysis(
     return response.result
 
 
+@with_api_client
+def build_analysis(
+    api_client: ApiClient,
+    image: str,
+    base_image: str,
+    buildlog: str,
+    *,
+    environment_type: str,
+    registry_user: str = None,
+    registry_password: str = None,
+    verify_tls: bool = True,
+    nowait: bool = False,
+    force: bool = False,
+    debug: bool = False,
+) -> typing.Union[typing.Dict, str]:
+    """Submit a build for analysis to Thoth."""
+    if not base_image and not base_image:
+        raise ValueError("No build info provided")
+
+    api_instance = BuildAnalysisApi(api_client)
+    if not image:
+        response = api_instance.post_build(
+            log_info=buildlog, base_image=base_image, debug=debug, verify_tls=verify_tls, force=force
+        )
+    if registry_user or registry_password:
```

Comment:
This branching does not look correct to me. There can happen 2 requests?

Reply:
I will assume here that you are referencing to the outgoing 2 different API calls.

According to me, we need different API calls to be triggered suffice different scenarios.
The two different scenarios:
- The build is successful in the namespace we want BW to submit  **output Imagestream**, **base image stream**, and **Build log**.
- The build is unsuccessful in the namespace we want BW  to submit **base Imagestream**, and **build log**.

If this is not you were referring to. My apologies, can you rephrase the comment :) @fridex

Reply:
What about providing just 1 endpoint which will act based on the input?

* if all 3 (output imagestream, base image stream and build log) are provided
* if any 2 are provided
* if just one is provided

It will ensure that data are consistent and success/failure can be a flag. Might be easier to maintain the logic (maybe)?

BTW, as Christoph pointed out in user-api PR - it could worth to have output imagestream, base image stream, and build log part of the JSON input and have others parameters (from API design POV).